### ISSUE TITLE
[BUG] Pre-lease 2.2.12 + Release 2.2.15: Graph View works but does not update itself when changes made.

### ISSUE THREAD

**Issue by DX5536** [YarnSpinnerTool/VSCodeExtension#11]
Hello,

First, I would like to thank for the fix of the graph view (but only if I uninstall **Yarn Spinner Language Server**, thou I'm sure this was intended). I have found a new issue this time: 

- In Graph View I can click "Add Node" and this is also shown in the .yarn file. However, the Graph View itself doesn't update with the new changes. I have to close it and re-open to see the newly added node in the graph.

I have taken a video showcasing the issue here:

https://user-images.githubusercontent.com/49788345/174085435-a84065bf-d587-4dab-9eec-d2c929e92300.mp4

Here, I try to rename the newly added node. No changes are visible on Graph View and I got an error:

https://user-images.githubusercontent.com/49788345/174086104-36af7ccd-6c8c-4464-bb94-a70bbc0b9154.mp4

(You can see I have uninstalled **Yarn Spinner Language Server** in the left Extension window).
.

**Comment by DX5536** [comment]
With the release of v.2.2.15, this bug is sadly not fixed but seems to create a new problem: The graph now not even update after closing and open it again like Pre-release 2.2.12. Syntax didn't work properly ("title: Node" is not colored). Language Server has to be disabled.

https://user-images.githubusercontent.com/49788345/174753299-744f9eb5-a8a3-46f2-b971-0a607de3c6ee.mp4

**Comment by McJones** [comment]
I have been doing some testing around this and was able to somewhat reproduce this but it required an absolute hack to make reliably break that literally cannot be part of the cause, but it did give a lead.
If you could run some steps to see what's going on with the extension that would be very helpful in working out a fix.

1. In the extension settings in VS Code, so `Settings` -> `Extensions` -> `Yarn Spinner` set the `Yarnspinner > Trace:Server` dropdown to be `verbose`.
(You may need to restart VS Code once that is done.)
2. Open up the Output window, so `View` -> `Output`, and in the dropdown on the right side of the Output window select `Yarn Spinner`, this will now log all communication between vscode and the extension.
3. Once that is done if you can make a new yarn file and then open the graph view for that file.
4. Then make a new node and move it around a bit.
5. Copy __ALL__ the text from the Output window (it will be a lot) and put that here/online somewhere you can link to

My working theory based on some testing is the language server is losing connection to the extension preventing any changes in the either the text or graph view being sent over to the other.

**Comment by DX5536** [comment]
```
[Trace - 00:49:49] Sending notification 'textDocument/didOpen'.
Params: {
    "textDocument": {
        "uri": "file:///c%3A/Users/Hai%20Lam/Desktop/test.yarn",
        "languageId": "yarnspinner",
        "version": 5,
        "text": ""
    }
}


[Trace - 00:49:49] Sending request 'textDocument/documentSymbol - (26)'.
Params: {
    "textDocument": {
        "uri": "file:///c%3A/Users/Hai%20Lam/Desktop/test.yarn"
    }
}


[Trace - 00:49:49] Sending notification '$/cancelRequest'.
Params: {
    "id": 26
}


[Trace - 00:49:49] Sending request 'textDocument/documentSymbol - (27)'.
Params: {
    "textDocument": {
        "uri": "file:///c%3A/Users/Hai%20Lam/Desktop/test.yarn"
    }
}


[Trace - 00:49:49] Sending request 'textDocument/semanticTokens/full - (28)'.
Params: {
    "textDocument": {
        "uri": "file:///c%3A/Users/Hai%20Lam/Desktop/test.yarn"
    }
}


line 0:-1 mismatched input '<EOF>' expecting {ID, '#'}
[Trace - 00:49:49] Received notification 'textDocument/yarnSpinner/didChangeNodes'.
Params: {
    "uri": "file:///c:/Users/Hai%20Lam/Desktop/test.yarn",
    "nodes": []
}


[Trace - 00:49:49] Received notification 'textDocument/publishDiagnostics'.
Params: {
    "uri": "file:///c:/Users/Hai%20Lam/Desktop/UE_GitKraken_Repo/TheNeighbourhood_SoSe22/TheNeighbourhood/Assets/YarnSpinner/Item_Thoughts_Script.yarn",
    "diagnostics": []
}


[Trace - 00:49:49] Received notification 'textDocument/publishDiagnostics'.
Params: {
    "uri": "file:///c:/Users/Hai%20Lam/Desktop/UE_GitKraken_Repo/TheNeighbourhood_SoSe22/TheNeighbourhood/Assets/YarnSpinner/NPC_Bird_Mia_Script.yarn",
    "diagnostics": []
}


[Trace - 00:49:49] Received notification 'textDocument/publishDiagnostics'.
Params: {
    "uri": "file:///c:/Users/Hai%20Lam/Desktop/UE_GitKraken_Repo/TheNeighbourhood_SoSe22/TheNeighbourhood/Assets/YarnSpinner/NPC_Grandparents_Script.yarn",
    "diagnostics": []
}


[Trace - 00:49:49] Received notification 'textDocument/publishDiagnostics'.
Params: {
    "uri": "file:///c:/Users/Hai%20Lam/Desktop/UE_GitKraken_Repo/TheNeighbourhood_SoSe22/TheNeighbourhood/Assets/YarnSpinner/NPC_Rabbit_Caro_Script.yarn",
    "diagnostics": []
}


[Trace - 00:49:49] Received notification 'textDocument/publishDiagnostics'.
Params: {
    "uri": "file:///c:/Users/Hai%20Lam/Desktop/UE_GitKraken_Repo/TheNeighbourhood_SoSe22/TheNeighbourhood/Assets/YarnSpinner/NPC_Squirrel_Nora_Script.yarn",
    "diagnostics": []
}


[Trace - 00:49:49] Received notification 'textDocument/publishDiagnostics'.
Params: {
    "uri": "file:///c:/Users/Hai%20Lam/Desktop/UE_GitKraken_Repo/TheNeighbourhood_SoSe22/TheNeighbourhood/Assets/YarnSpinner/NPC_Wolf_Tom_Script.yarn",
    "diagnostics": []
}


[Trace - 00:49:49] Received notification 'textDocument/publishDiagnostics'.
Params: {
    "uri": "file:///c:/Users/Hai%20Lam/Desktop/UE_GitKraken_Repo/TheNeighbourhood_SoSe22/TheNeighbourhood/Assets/YarnSpinner/OLD_OUTDATED_YarnScripts/MeetingGrandparentsDialogue.yarn",
    "diagnostics": []
}


[Trace - 00:49:49] Received notification 'textDocument/publishDiagnostics'.
Params: {
    "uri": "file:///c:/Users/Hai%20Lam/Desktop/UE_GitKraken_Repo/TheNeighbourhood_SoSe22/TheNeighbourhood/Assets/YarnSpinner/OLD_OUTDATED_YarnScripts/Meeting_Grandparents_Script.yarn",
    "diagnostics": []
}


[Trace - 00:49:49] Received notification 'textDocument/publishDiagnostics'.
Params: {
    "uri": "file:///c:/Users/Hai%20Lam/Desktop/UE_GitKraken_Repo/TheNeighbourhood_SoSe22/TheNeighbourhood/Assets/Samples/Yarn%20Spinner/2.2.1/3D%20Speech%20Bubble/Dialogue/3dSampleDialogue.yarn",
    "diagnostics": []
}


[Trace - 00:49:49] Received notification 'textDocument/publishDiagnostics'.
Params: {
    "uri": "file:///c:/Users/Hai%20Lam/Desktop/UE_GitKraken_Repo/TheNeighbourhood_SoSe22/TheNeighbourhood/Assets/Samples/Yarn%20Spinner/2.2.1/Visual%20Novel/Dialogue/VNExampleDialogue.yarn",
    "diagnostics": []
}


[Trace - 00:49:49] Received notification 'textDocument/publishDiagnostics'.
Params: {
    "uri": "file:///c:/Users/Hai%20Lam/Desktop/Sample.yarn",
    "diagnostics": [
        {
            "range": {
                "start": {
                    "line": 8,
                    "character": 8
                },
                "end": {
                    "line": 8,
                    "character": 18
                }
            },
            "severity": 2,
            "code": "YRNMsngCmdDef",
            "message": "Could not find function definition for JumpToNode",
            "data": {
                "Item1": "JumpToNode",
                "Item2": false
            }
        },
        {
            "range": {
                "start": {
                    "line": 6,
                    "character": 2
                },
                "end": {
                    "line": 6,
                    "character": 22
                }
            },
            "severity": 2,
            "code": "YRNMsngCmdDef",
            "message": "Could not find command definition for SentenceBuilderStart",
            "data": {
                "Item1": "SentenceBuilderStart",
                "Item2": true
            }
        }
    ]
}


[Trace - 00:49:49] Received notification 'textDocument/publishDiagnostics'.
Params: {
    "uri": "file:///c:/Users/Hai%20Lam/Desktop/test.yarn",
    "diagnostics": [
        {
            "range": {
                "start": {
                    "line": -1,
                    "character": -1
                },
                "end": {
                    "line": -1,
                    "character": 0
                }
            },
            "severity": 1,
            "source": "file:///c:/Users/Hai Lam/Desktop/test.yarn",
            "message": "Unexpected \"<EOF>\" while reading a dialogue"
        }
    ]
}


[Trace - 00:49:49] Received response 'textDocument/documentSymbol - (26)' in 27ms.
Result: []


[Trace - 00:49:49] Received response 'textDocument/documentSymbol - (27)' in 26ms.
Result: []


[Trace - 00:49:49] Received response 'textDocument/semanticTokens/full - (28)' in 26ms.
Result: {
    "resultId": "401e8978-e919-4345-8088-edc6361283a5",
    "data": []
}


[Trace - 00:49:49] Sending request 'textDocument/codeLens - (29)'.
Params: {
    "textDocument": {
        "uri": "file:///c%3A/Users/Hai%20Lam/Desktop/test.yarn"
    }
}


[Trace - 00:49:49] Received response 'textDocument/codeLens - (29)' in 1ms.
Result: []


[Trace - 00:49:53] Sending request 'workspace/executeCommand - (30)'.
Params: {
    "command": "yarnspinner.list-nodes",
    "arguments": [
        "c:\\Users\\Hai Lam\\Desktop\\test.yarn"
    ]
}


[Trace - 00:49:53] Received response 'workspace/executeCommand - (30)' in 2ms.
Result: []


[Trace - 00:49:54] Sending request 'workspace/executeCommand - (31)'.
Params: {
    "command": "yarnspinner.create-node",
    "arguments": [
        "c:\\Users\\Hai Lam\\Desktop\\test.yarn",
        {
            "position": "0,1"
        }
    ]
}


[Trace - 00:49:54] Received response 'workspace/executeCommand - (31)' in 2ms.
Result: {
    "textDocument": {
        "version": null,
        "uri": "file:///c:/Users/Hai%20Lam/Desktop/test.yarn"
    },
    "edits": [
        {
            "range": {
                "start": {
                    "line": 0,
                    "character": 0
                },
                "end": {
                    "line": 0,
                    "character": 0
                }
            },
            "newText": "title: Node\r\nposition: 0,1\r\n---\r\n\r\n===\r\n"
        }
    ]
}


[Trace - 00:49:55] Sending notification 'textDocument/didChange'.
Params: {
    "textDocument": {
        "uri": "file:///c%3A/Users/Hai%20Lam/Desktop/test.yarn",
        "version": 6
    },
    "contentChanges": [
        {
            "text": "title: Node\r\nposition: 0,1\r\n---\r\n\r\n===\r\n"
        }
    ]
}


[Trace - 00:49:55] Received notification 'textDocument/yarnSpinner/didChangeNodes'.
Params: {
    "uri": "file:///c:/Users/Hai%20Lam/Desktop/test.yarn",
    "nodes": [
        {
            "title": "Node",
            "bodyStartLine": 3,
            "bodyEndLine": 3,
            "headerStartLine": 0,
            "headers": [
                {
                    "key": "title",
                    "value": "Node"
                },
                {
                    "key": "position",
                    "value": "0,1"
                }
            ],
            "jumps": [],
            "hasTitle": true
        }
    ]
}


[Trace - 00:49:55] Received notification 'textDocument/publishDiagnostics'.
Params: {
    "uri": "file:///c:/Users/Hai%20Lam/Desktop/UE_GitKraken_Repo/TheNeighbourhood_SoSe22/TheNeighbourhood/Assets/YarnSpinner/Item_Thoughts_Script.yarn",
    "diagnostics": []
}


[Trace - 00:49:55] Received notification 'textDocument/publishDiagnostics'.
Params: {
    "uri": "file:///c:/Users/Hai%20Lam/Desktop/UE_GitKraken_Repo/TheNeighbourhood_SoSe22/TheNeighbourhood/Assets/YarnSpinner/NPC_Bird_Mia_Script.yarn",
    "diagnostics": []
}


[Trace - 00:49:55] Received notification 'textDocument/publishDiagnostics'.
Params: {
    "uri": "file:///c:/Users/Hai%20Lam/Desktop/UE_GitKraken_Repo/TheNeighbourhood_SoSe22/TheNeighbourhood/Assets/YarnSpinner/NPC_Grandparents_Script.yarn",
    "diagnostics": []
}


[Trace - 00:49:55] Received notification 'textDocument/publishDiagnostics'.
Params: {
    "uri": "file:///c:/Users/Hai%20Lam/Desktop/UE_GitKraken_Repo/TheNeighbourhood_SoSe22/TheNeighbourhood/Assets/YarnSpinner/NPC_Rabbit_Caro_Script.yarn",
    "diagnostics": []
}


[Trace - 00:49:55] Received notification 'textDocument/publishDiagnostics'.
Params: {
    "uri": "file:///c:/Users/Hai%20Lam/Desktop/UE_GitKraken_Repo/TheNeighbourhood_SoSe22/TheNeighbourhood/Assets/YarnSpinner/NPC_Squirrel_Nora_Script.yarn",
    "diagnostics": []
}


[Trace - 00:49:55] Received notification 'textDocument/publishDiagnostics'.
Params: {
    "uri": "file:///c:/Users/Hai%20Lam/Desktop/UE_GitKraken_Repo/TheNeighbourhood_SoSe22/TheNeighbourhood/Assets/YarnSpinner/NPC_Wolf_Tom_Script.yarn",
    "diagnostics": []
}


[Trace - 00:49:55] Received notification 'textDocument/publishDiagnostics'.
Params: {
    "uri": "file:///c:/Users/Hai%20Lam/Desktop/UE_GitKraken_Repo/TheNeighbourhood_SoSe22/TheNeighbourhood/Assets/YarnSpinner/OLD_OUTDATED_YarnScripts/MeetingGrandparentsDialogue.yarn",
    "diagnostics": []
}


[Trace - 00:49:55] Received notification 'textDocument/publishDiagnostics'.
Params: {
    "uri": "file:///c:/Users/Hai%20Lam/Desktop/UE_GitKraken_Repo/TheNeighbourhood_SoSe22/TheNeighbourhood/Assets/YarnSpinner/OLD_OUTDATED_YarnScripts/Meeting_Grandparents_Script.yarn",
    "diagnostics": []
}


[Trace - 00:49:55] Received notification 'textDocument/publishDiagnostics'.
Params: {
    "uri": "file:///c:/Users/Hai%20Lam/Desktop/UE_GitKraken_Repo/TheNeighbourhood_SoSe22/TheNeighbourhood/Assets/Samples/Yarn%20Spinner/2.2.1/3D%20Speech%20Bubble/Dialogue/3dSampleDialogue.yarn",
    "diagnostics": []
}


[Trace - 00:49:55] Received notification 'textDocument/publishDiagnostics'.
Params: {
    "uri": "file:///c:/Users/Hai%20Lam/Desktop/UE_GitKraken_Repo/TheNeighbourhood_SoSe22/TheNeighbourhood/Assets/Samples/Yarn%20Spinner/2.2.1/Visual%20Novel/Dialogue/VNExampleDialogue.yarn",
    "diagnostics": []
}


[Trace - 00:49:55] Received notification 'textDocument/publishDiagnostics'.
Params: {
    "uri": "file:///c:/Users/Hai%20Lam/Desktop/Sample.yarn",
    "diagnostics": [
        {
            "range": {
                "start": {
                    "line": 8,
                    "character": 8
                },
                "end": {
                    "line": 8,
                    "character": 18
                }
            },
            "severity": 2,
            "code": "YRNMsngCmdDef",
            "message": "Could not find function definition for JumpToNode",
            "data": {
                "Item1": "JumpToNode",
                "Item2": false
            }
        },
        {
            "range": {
                "start": {
                    "line": 6,
                    "character": 2
                },
                "end": {
                    "line": 6,
                    "character": 22
                }
            },
            "severity": 2,
            "code": "YRNMsngCmdDef",
            "message": "Could not find command definition for SentenceBuilderStart",
            "data": {
                "Item1": "SentenceBuilderStart",
                "Item2": true
            }
        }
    ]
}


[Trace - 00:49:55] Received notification 'textDocument/publishDiagnostics'.
Params: {
    "uri": "file:///c:/Users/Hai%20Lam/Desktop/test.yarn",
    "diagnostics": []
}


[Trace - 00:49:55] Sending request 'textDocument/codeLens - (32)'.
Params: {
    "textDocument": {
        "uri": "file:///c%3A/Users/Hai%20Lam/Desktop/test.yarn"
    }
}


[Trace - 00:49:55] Received response 'textDocument/codeLens - (32)' in 1ms.
Result: [
    {
        "range": {
            "start": {
                "line": 0,
                "character": 7
            },
            "end": {
                "line": 0,
                "character": 11
            }
        },
        "command": {
            "title": "0 references",
            "command": "yarn.showReferences",
            "arguments": [
                {
                    "line": 0,
                    "character": 7
                },
                [
                    {
                        "uri": "file:///c:/Users/Hai%20Lam/Desktop/test.yarn",
                        "range": {
                            "start": {
                                "line": 0,
                                "character": 7
                            },
                            "end": {
                                "line": 0,
                                "character": 11
                            }
                        }
                    }
                ]
            ]
        },
        "data": {}
    }
]


[Trace - 00:49:55] Sending request 'textDocument/semanticTokens/full - (33)'.
Params: {
    "textDocument": {
        "uri": "file:///c%3A/Users/Hai%20Lam/Desktop/test.yarn"
    }
}


[Trace - 00:49:55] Received response 'textDocument/semanticTokens/full - (33)' in 1ms.
Result: {
    "resultId": "9d3c9a46-4232-4439-addc-46994c6e9613",
    "data": [
        0,
        0,
        5,
        9,
        0,
        0,
        7,
        4,
        2,
        0,
        0,
        4,
        2,
        17,
        0,
        1,
        0,
        8,
        9,
        0,
        0,
        10,
        3,
        18,
        0,
        0,
        3,
        2,
        17,
        0,
        1,
        3,
        2,
        17,
        0,
        1,
        0,
        2,
        17,
        0,
        1,
        3,
        2,
        17,
        0
    ]
}


[Trace - 00:49:55] Sending request 'textDocument/documentSymbol - (34)'.
Params: {
    "textDocument": {
        "uri": "file:///c%3A/Users/Hai%20Lam/Desktop/test.yarn"
    }
}


[Trace - 00:49:55] Received response 'textDocument/documentSymbol - (34)' in 1ms.
Result: [
    {
        "name": "Node",
        "kind": 19,
        "range": {
            "start": {
                "line": 0,
                "character": 0
            },
            "end": {
                "line": 4,
                "character": 3
            }
        },
        "selectionRange": {
            "start": {
                "line": 0,
                "character": 0
            },
            "end": {
                "line": 0,
                "character": 5
            }
        },
        "children": [
            {
                "name": "title",
                "detail": "Node",
                "kind": 7,
                "range": {
                    "start": {
                        "line": 0,
                        "character": 0
                    },
                    "end": {
                        "line": 0,
                        "character": 11
                    }
                },
                "selectionRange": {
                    "start": {
                        "line": 0,
                        "character": 0
                    },
                    "end": {
                        "line": 0,
                        "character": 11
                    }
                }
            },
            {
                "name": "position",
                "detail": "0,1",
                "kind": 7,
                "range": {
                    "start": {
                        "line": 1,
                        "character": 0
                    },
                    "end": {
                        "line": 1,
                        "character": 13
                    }
                },
                "selectionRange": {
                    "start": {
                        "line": 1,
                        "character": 0
                    },
                    "end": {
                        "line": 1,
                        "character": 13
                    }
                }
            }
        ]
    }
]


[Trace - 00:50:01] Sending notification 'textDocument/didSave'.
Params: {
    "textDocument": {
        "uri": "file:///c%3A/Users/Hai%20Lam/Desktop/test.yarn"
    },
    "text": "title: Node\r\nposition: 0,1\r\n---\r\n\r\n===\r\n"
}


[Trace - 00:50:01] Sending request 'textDocument/codeLens - (35)'.
Params: {
    "textDocument": {
        "uri": "file:///c%3A/Users/Hai%20Lam/Desktop/test.yarn"
    }
}


[Trace - 00:50:01] Received response 'textDocument/codeLens - (35)' in 2ms.
Result: [
    {
        "range": {
            "start": {
                "line": 0,
                "character": 7
            },
            "end": {
                "line": 0,
                "character": 11
            }
        },
        "command": {
            "title": "0 references",
            "command": "yarn.showReferences",
            "arguments": [
                {
                    "line": 0,
                    "character": 7
                },
                [
                    {
                        "uri": "file:///c:/Users/Hai%20Lam/Desktop/test.yarn",
                        "range": {
                            "start": {
                                "line": 0,
                                "character": 7
                            },
                            "end": {
                                "line": 0,
                                "character": 11
                            }
                        }
                    }
                ]
            ]
        },
        "data": {}
    }
]


[Trace - 00:50:02] Sending request 'workspace/executeCommand - (36)'.
Params: {
    "command": "yarnspinner.list-nodes",
    "arguments": [
        "c:\\Users\\Hai Lam\\Desktop\\test.yarn"
    ]
}


[Trace - 00:50:02] Received response 'workspace/executeCommand - (36)' in 2ms.
Result: [
    {
        "title": "Node",
        "bodyStartLine": 3,
        "bodyEndLine": 3,
        "headerStartLine": 0,
        "headers": [
            {
                "key": "title",
                "value": "Node"
            },
            {
                "key": "position",
                "value": "0,1"
            }
        ],
        "jumps": [],
        "hasTitle": true
    }
]


[Trace - 00:50:06] Sending request 'workspace/executeCommand - (37)'.
Params: {
    "command": "yarnspinner.update-node-header",
    "arguments": [
        "c:\\Users\\Hai Lam\\Desktop\\test.yarn",
        "Node",
        "position",
        "-60,31"
    ]
}


[Trace - 00:50:06] Received response 'workspace/executeCommand - (37)' in 2ms.
Result: {
    "textDocument": {
        "version": null,
        "uri": "file:///c:/Users/Hai%20Lam/Desktop/test.yarn"
    },
    "edits": [
        {
            "range": {
                "start": {
                    "line": 1,
                    "character": 0
                },
                "end": {
                    "line": 1,
                    "character": 13
                }
            },
            "newText": "position: -60,31"
        }
    ]
}


[Trace - 00:50:06] Sending notification 'textDocument/didChange'.
Params: {
    "textDocument": {
        "uri": "file:///c%3A/Users/Hai%20Lam/Desktop/test.yarn",
        "version": 7
    },
    "contentChanges": [
        {
            "text": "title: Node\r\nposition: -60,31\r\n---\r\n\r\n===\r\n"
        }
    ]
}


[Trace - 00:50:06] Received notification 'textDocument/yarnSpinner/didChangeNodes'.
Params: {
    "uri": "file:///c:/Users/Hai%20Lam/Desktop/test.yarn",
    "nodes": [
        {
            "title": "Node",
            "bodyStartLine": 3,
            "bodyEndLine": 3,
            "headerStartLine": 0,
            "headers": [
                {
                    "key": "title",
                    "value": "Node"
                },
                {
                    "key": "position",
                    "value": "-60,31"
                }
            ],
            "jumps": [],
            "hasTitle": true
        }
    ]
}


[Trace - 00:50:06] Received notification 'textDocument/publishDiagnostics'.
Params: {
    "uri": "file:///c:/Users/Hai%20Lam/Desktop/UE_GitKraken_Repo/TheNeighbourhood_SoSe22/TheNeighbourhood/Assets/YarnSpinner/Item_Thoughts_Script.yarn",
    "diagnostics": []
}


[Trace - 00:50:06] Received notification 'textDocument/publishDiagnostics'.
Params: {
    "uri": "file:///c:/Users/Hai%20Lam/Desktop/UE_GitKraken_Repo/TheNeighbourhood_SoSe22/TheNeighbourhood/Assets/YarnSpinner/NPC_Bird_Mia_Script.yarn",
    "diagnostics": []
}


[Trace - 00:50:06] Received notification 'textDocument/publishDiagnostics'.
Params: {
    "uri": "file:///c:/Users/Hai%20Lam/Desktop/UE_GitKraken_Repo/TheNeighbourhood_SoSe22/TheNeighbourhood/Assets/YarnSpinner/NPC_Grandparents_Script.yarn",
    "diagnostics": []
}


[Trace - 00:50:06] Received notification 'textDocument/publishDiagnostics'.
Params: {
    "uri": "file:///c:/Users/Hai%20Lam/Desktop/UE_GitKraken_Repo/TheNeighbourhood_SoSe22/TheNeighbourhood/Assets/YarnSpinner/NPC_Rabbit_Caro_Script.yarn",
    "diagnostics": []
}


[Trace - 00:50:06] Received notification 'textDocument/publishDiagnostics'.
Params: {
    "uri": "file:///c:/Users/Hai%20Lam/Desktop/UE_GitKraken_Repo/TheNeighbourhood_SoSe22/TheNeighbourhood/Assets/YarnSpinner/NPC_Squirrel_Nora_Script.yarn",
    "diagnostics": []
}


[Trace - 00:50:06] Received notification 'textDocument/publishDiagnostics'.
Params: {
    "uri": "file:///c:/Users/Hai%20Lam/Desktop/UE_GitKraken_Repo/TheNeighbourhood_SoSe22/TheNeighbourhood/Assets/YarnSpinner/NPC_Wolf_Tom_Script.yarn",
    "diagnostics": []
}


[Trace - 00:50:06] Received notification 'textDocument/publishDiagnostics'.
Params: {
    "uri": "file:///c:/Users/Hai%20Lam/Desktop/UE_GitKraken_Repo/TheNeighbourhood_SoSe22/TheNeighbourhood/Assets/YarnSpinner/OLD_OUTDATED_YarnScripts/MeetingGrandparentsDialogue.yarn",
    "diagnostics": []
}


[Trace - 00:50:06] Received notification 'textDocument/publishDiagnostics'.
Params: {
    "uri": "file:///c:/Users/Hai%20Lam/Desktop/UE_GitKraken_Repo/TheNeighbourhood_SoSe22/TheNeighbourhood/Assets/YarnSpinner/OLD_OUTDATED_YarnScripts/Meeting_Grandparents_Script.yarn",
    "diagnostics": []
}


[Trace - 00:50:06] Received notification 'textDocument/publishDiagnostics'.
Params: {
    "uri": "file:///c:/Users/Hai%20Lam/Desktop/UE_GitKraken_Repo/TheNeighbourhood_SoSe22/TheNeighbourhood/Assets/Samples/Yarn%20Spinner/2.2.1/3D%20Speech%20Bubble/Dialogue/3dSampleDialogue.yarn",
    "diagnostics": []
}


[Trace - 00:50:06] Received notification 'textDocument/publishDiagnostics'.
Params: {
    "uri": "file:///c:/Users/Hai%20Lam/Desktop/UE_GitKraken_Repo/TheNeighbourhood_SoSe22/TheNeighbourhood/Assets/Samples/Yarn%20Spinner/2.2.1/Visual%20Novel/Dialogue/VNExampleDialogue.yarn",
    "diagnostics": []
}


[Trace - 00:50:06] Received notification 'textDocument/publishDiagnostics'.
Params: {
    "uri": "file:///c:/Users/Hai%20Lam/Desktop/Sample.yarn",
    "diagnostics": [
        {
            "range": {
                "start": {
                    "line": 8,
                    "character": 8
                },
                "end": {
                    "line": 8,
                    "character": 18
                }
            },
            "severity": 2,
            "code": "YRNMsngCmdDef",
            "message": "Could not find function definition for JumpToNode",
            "data": {
                "Item1": "JumpToNode",
                "Item2": false
            }
        },
        {
            "range": {
                "start": {
                    "line": 6,
                    "character": 2
                },
                "end": {
                    "line": 6,
                    "character": 22
                }
            },
            "severity": 2,
            "code": "YRNMsngCmdDef",
            "message": "Could not find command definition for SentenceBuilderStart",
            "data": {
                "Item1": "SentenceBuilderStart",
                "Item2": true
            }
        }
    ]
}


[Trace - 00:50:06] Received notification 'textDocument/publishDiagnostics'.
Params: {
    "uri": "file:///c:/Users/Hai%20Lam/Desktop/test.yarn",
    "diagnostics": []
}


[Trace - 00:50:06] Sending request 'textDocument/codeLens - (38)'.
Params: {
    "textDocument": {
        "uri": "file:///c%3A/Users/Hai%20Lam/Desktop/test.yarn"
    }
}


[Trace - 00:50:06] Received response 'textDocument/codeLens - (38)' in 1ms.
Result: [
    {
        "range": {
            "start": {
                "line": 0,
                "character": 7
            },
            "end": {
                "line": 0,
                "character": 11
            }
        },
        "command": {
            "title": "0 references",
            "command": "yarn.showReferences",
            "arguments": [
                {
                    "line": 0,
                    "character": 7
                },
                [
                    {
                        "uri": "file:///c:/Users/Hai%20Lam/Desktop/test.yarn",
                        "range": {
                            "start": {
                                "line": 0,
                                "character": 7
                            },
                            "end": {
                                "line": 0,
                                "character": 11
                            }
                        }
                    }
                ]
            ]
        },
        "data": {}
    }
]


[Trace - 00:50:06] Sending request 'textDocument/semanticTokens/full - (39)'.
Params: {
    "textDocument": {
        "uri": "file:///c%3A/Users/Hai%20Lam/Desktop/test.yarn"
    }
}


[Trace - 00:50:06] Received response 'textDocument/semanticTokens/full - (39)' in 1ms.
Result: {
    "resultId": "415ca388-8d28-4139-a17b-7c19297707f1",
    "data": [
        0,
        0,
        5,
        9,
        0,
        0,
        7,
        4,
        2,
        0,
        0,
        4,
        2,
        17,
        0,
        1,
        0,
        8,
        9,
        0,
        0,
        10,
        6,
        18,
        0,
        0,
        6,
        2,
        17,
        0,
        1,
        3,
        2,
        17,
        0,
        1,
        0,
        2,
        17,
        0,
        1,
        3,
        2,
        17,
        0
    ]
}


[Trace - 00:50:06] Sending request 'textDocument/documentSymbol - (40)'.
Params: {
    "textDocument": {
        "uri": "file:///c%3A/Users/Hai%20Lam/Desktop/test.yarn"
    }
}


[Trace - 00:50:06] Received response 'textDocument/documentSymbol - (40)' in 1ms.
Result: [
    {
        "name": "Node",
        "kind": 19,
        "range": {
            "start": {
                "line": 0,
                "character": 0
            },
            "end": {
                "line": 4,
                "character": 3
            }
        },
        "selectionRange": {
            "start": {
                "line": 0,
                "character": 0
            },
            "end": {
                "line": 0,
                "character": 5
            }
        },
        "children": [
            {
                "name": "title",
                "detail": "Node",
                "kind": 7,
                "range": {
                    "start": {
                        "line": 0,
                        "character": 0
                    },
                    "end": {
                        "line": 0,
                        "character": 11
                    }
                },
                "selectionRange": {
                    "start": {
                        "line": 0,
                        "character": 0
                    },
                    "end": {
                        "line": 0,
                        "character": 11
                    }
                }
            },
            {
                "name": "position",
                "detail": "-60,31",
                "kind": 7,
                "range": {
                    "start": {
                        "line": 1,
                        "character": 0
                    },
                    "end": {
                        "line": 1,
                        "character": 16
                    }
                },
                "selectionRange": {
                    "start": {
                        "line": 1,
                        "character": 0
                    },
                    "end": {
                        "line": 1,
                        "character": 16
                    }
                }
            }
        ]
    }
]
```

So here the output messages. I also tried to record my steps as you instructed (with a little test before hand).

https://user-images.githubusercontent.com/49788345/180095937-441fb29e-9056-4eee-be7f-d46de6aa8554.mp4

**Comment by desplesda** [comment]
Fixed in cc8f3d3b724da8f7e3ba98cecf754bb9461268ad. Thank you for the detailed report!
